IncludeToolbox/Formatter/IncludeFormatter.cs

```diff
@@ -47,16 +56,44 @@ public static string FormatPath(string absoluteIncludeFilename, FormatterOptions
         /// <summary>
         /// Formats the paths of a given list of include line info.
         /// </summary>
-        private static void FormatPaths(IEnumerable<IncludeLineInfo> lines, FormatterOptionsPage.PathMode pathformat, IEnumerable<string> includeDirectories)
+        private static void FormatPaths(IEnumerable<IncludeLineInfo> lines,
+                                        FormatterOptionsPage.PathMode pathformat,
+                                        FormatterOptionsPage.IgnoreFileRelativeMode ignoreFileRelativeMode,
+                                        IEnumerable<string> includeDirectories,
+                                        string documentDir,
+                                        string includeRootDirectory = null)
         {
             if (pathformat == FormatterOptionsPage.PathMode.Unchanged)
                 return;
 
             foreach (var line in lines)
             {
-                string absoluteIncludeDir = line.TryResolveInclude(includeDirectories, out bool resolvedPath);
+                string absoluteIncludePath = line.TryResolveInclude(includeDirectories, out bool resolvedPath);
                 if (resolvedPath)
-                    line.IncludeContent = FormatPath(absoluteIncludeDir, pathformat, includeDirectories) ?? line.IncludeContent;
+                {
+                    if (pathformat == FormatterOptionsPage.PathMode.Absolute_FromParentDirWithFile &&
+                        includeRootDirectory != null &&
+                        !absoluteIncludePath.StartsWith(includeRootDirectory))
+                    {
+                        continue;
+                    }
+
+                    var currentPathFormat = pathformat;
+                    var currentIncludeRootDirectory = includeRootDirectory;
+                    string includeFilename = Path.GetFileName(absoluteIncludePath);
+                    if ((ignoreFileRelativeMode == FormatterOptionsPage.IgnoreFileRelativeMode.InSameDirectory &&
+                         Path.Combine(documentDir, includeFilename) == absoluteIncludePath) ||
+                        (ignoreFileRelativeMode == FormatterOptionsPage.IgnoreFileRelativeMode.InSameOrSubDirectory &&
+                         absoluteIncludePath.StartsWith(documentDir)))
+                    {
+                        currentPathFormat = FormatterOptionsPage.PathMode.Absolute_FromParentDirWithFile;
+                        currentIncludeRootDirectory = documentDir;
+                    }
+                    line.IncludeContent = FormatPath(absoluteIncludePath,
```

Comment:
Old version was guarded against FormatPath returning null. There might have been a good reason for that.

Reply:
It still is, just a few lines down because of how I formatted the parameters. (I dislike commenting on code in GitHub because I find inline comments hard to place well and hard to read -- I'd rather them show up in some kind of side bar so they don't split up the actual code!)